packages/jest-resolve/src/node_modules_paths.js

```diff
@@ -37,11 +38,17 @@ export default function nodeModulesPaths(
     prefix = '\\\\';
   }
 
-  const paths = [basedirAbs];
-  let parsed = path.parse(basedirAbs);
+  // The node resolution algorithm (as implemented by NodeJS
+  // and TypeScript) traverses parents of the physical path,
+  // not the symlinked path 
+  const physicalBasedir: string = fs.realpathSync(basedirAbs);
```

Comment:
Please use the native realpath.

https://github.com/facebook/jest/blob/cfb7ee107271f6b0d3921839699e313b9411c794/packages/jest-util/src/index.js#L35-L42

Reply:
we've had issues with casing on windows using `fs.realpathSync`

Reply:
Thanks